src/java/org/apache/cassandra/index/sai/disk/v1/SegmentBuilder.java

```diff
@@ -78,8 +78,10 @@ public abstract class SegmentBuilder
 {
     private static final Logger logger = LoggerFactory.getLogger(SegmentBuilder.class);
 
-    /** for parallelism within a single compaction */
-    public static final ExecutorService compactionExecutor = new DebuggableThreadPoolExecutor(Runtime.getRuntime().availableProcessors(),
+    /** for parallelism within a single compaction
+     *  see comments to JVector PhysicalCoreExecutor -- HT tends to cause contention for the SIMD units
+     */
+    public static final ExecutorService compactionExecutor = new DebuggableThreadPoolExecutor(Runtime.getRuntime().availableProcessors() / 2,
```

Comment:
what about making this configurable ? in case we need to rollback

Reply:
fwiw, I copied this logic from:

https://github.com/datastax/cassandra/blob/61f57a602424e053039f90fd2479f826c0e2d9df/src/java/org/apache/cassandra/index/sai/disk/vector/CompactionGraph.java#L104-L108

Are we looking to make all of these configurable?